var/spack/repos/builtin/packages/gnutls/package.py

```diff
@@ -88,3 +88,11 @@ def configure_args(self):
             ])
 
         return args
+
+    @property
+    def headers(self):
+        # GnuTLS headers are located inside prefix.include.gnutls and included
+        # as <gnutls/gnutls.h>, so we return a fake header in prefix.include.
```

Comment:
why do we do that? sorry, i don't follow the logic of not returning the actual header.

Reply:
ah, now I see, you want the include flags to be right, i.e. `-I prefix.include`.

I would actually not rush this PR and instead see if we can solve this properly.

@adamjstewart any ideas? I think for headers we mostly need just the folder. So maybe `HeaderList` needs to be updated to be able to store just a set of folders **without** headers? won't be a `list` per-se, of course...

Reply:
Yes, that is the goal. There are other packages that use similar layout, e.g. sundials.

Reply:
Why not just use `spec['gnutls'].prefix.include`? If `headers` isn't working for you, there's no need to force yourself to use it.

Reply:
@v-dobrev @adamjstewart A good solution to this might be to add an additional property to  `HeaderList` that says from which directory people usually start to include headers, and maintain it what it currently is: a list of header files. What I have in mind is something like:
```python
headers = spec['boost'].headers
headers.include_starts_at = 'boost'
headers.cpp_flags # Should give a single -I<prefix-to-boost>/include
```
Thoughts?

Reply:
Since we want to preserve the ability to sum `HeaderList`s, it will be necessary to classify the headers inside the `HeaderList` by e.g. their relative level of inclusion. The current behavior is level 0; one level up is level 1 (or -1), etc. It should be sufficient to add an optional parameter to the `HeaderList` constructor that specifies the (single) relative level for the given list of headers.

Reply:
> A good solution to this might be to add an additional property to HeaderList that says from which directory people usually start to include headers

👍